pkg/controller/clusterinstallation/controller_test.go

```diff
@@ -149,6 +110,45 @@ func TestReconcile(t *testing.T) {
 	}
 	defer c.Delete(context.TODO(), minioService)
 
+	// Create the ClusterInstallation object and expect the Reconcile and Deployment to be created
+	err = c.Create(context.TODO(), instance)
+	// The instance object may not be a valid object because it might be missing some required fields.
```

Comment:
Not sure how to parse this comment -- I guess it's instructions for when the e2e test fails in the future? Would it make sense to embed this into the failure message? Also, why doesn't this fail the test?

Reply:
me either :( not fully understand that as well

Reply:
After looking into the flapping tests I encountered on another branch. I think I figured out what they were trying to say here. This is my in-flight change:

```
	// Create the ClusterInstallation object.
	err = c.Create(context.TODO(), instance)
	// The instance object may not be a valid object because it might be missing
	// some required fields. We log the error if it IsInvalid() to help with
	// troubleshooting before ensuring the error is nil.
	if apierrors.IsInvalid(err) {
		t.Logf("failed to create object, object is invalid: %v", err)
	}
	g.Expect(err).NotTo(gomega.HaveOccurred())
```